pkg/deployment/agent.go

```diff
@@ -43,6 +44,11 @@ func (a *Agent) Get() *appsv1.DaemonSet {
 		if len(util.FindItem("--reporter.grpc.host-port=", args)) == 0 {
 			args = append(args, fmt.Sprintf("--reporter.grpc.host-port=dns:///%s.%s:14250", service.GetNameForHeadlessCollectorService(a.jaeger), a.jaeger.Namespace))
```

Comment:
There's something odd: this would generate a string like: `- --reporter.grpc.host-port=dns:///simple-prod-collector-headless.default:14250` . Yet, the cert is valid only for the service `simple-prod-collector`:

```console
$ kubectl get secrets jaeger-collector-tls -o=go-template='{{index .data "tls.crt"}}' | base64 -d | openssl x509 -text -noout
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 7784809259584540342 (0x6c0932c6a1b54eb6)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN = openshift-service-serving-signer@1582647274
        Validity
            Not Before: Feb 25 16:37:59 2020 GMT
            Not After : Feb 24 16:38:00 2022 GMT
        Subject: CN = simple-prod-collector.default.svc
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:d9:1c:19:e9:68:d4:17:fb:50:84:ba:32:7c:27:
                    ed:eb:f7:28:9e:76:e2:e9:5b:5f:73:12:31:24:49:
                    6f:1f:8d:1e:07:26:58:64:39:4a:48:1e:7d:f6:33:
                    02:38:17:6b:72:40:6c:4d:93:e6:d8:89:d8:07:f1:
                    40:46:fe:da:5e:dc:8c:de:21:1d:77:64:49:4f:8d:
                    3b:5b:56:70:da:a0:32:0a:48:75:d3:df:7f:77:72:
                    c4:42:09:28:db:70:93:dd:d9:cb:b6:d1:45:b5:7b:
                    38:39:a5:09:57:c5:cd:80:84:64:5f:66:2c:45:e1:
                    cc:53:f0:90:df:d9:09:a9:c8:b2:0d:19:c0:25:92:
                    f0:8a:35:1c:ab:3b:2e:21:06:c2:0e:57:38:90:95:
                    df:17:00:66:50:bb:80:b0:78:fe:12:35:bf:4b:26:
                    7e:a2:2c:38:d6:83:3f:87:42:c8:c7:12:bc:4a:fc:
                    13:01:ae:13:ba:94:9b:e1:de:37:e9:a6:d8:79:02:
                    57:21:8c:49:6e:4b:05:13:a7:b2:fd:17:08:3b:ce:
                    f4:b9:a1:35:39:df:6e:58:3b:57:47:b6:89:c3:95:
                    bb:65:0c:fd:e6:68:89:d9:63:09:82:4d:f1:00:97:
                    72:26:9e:34:60:5e:24:30:83:7b:b3:40:19:ce:8a:
                    58:9f
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                71:90:9D:90:2F:DB:63:5E:E6:53:24:50:FB:54:23:1E:DD:26:3A:5C
            X509v3 Authority Key Identifier: 
                keyid:F8:A9:03:4A:FD:7B:74:9F:D3:8E:F3:ED:B9:B7:D3:DC:E7:F5:85:5D

            X509v3 Subject Alternative Name: 
                DNS:simple-prod-collector.default.svc, DNS:simple-prod-collector.default.svc.cluster.local
            1.3.6.1.4.1.2312.17.100.2.1: 
                .$47b497a9-bbad-46a4-b5d9-6330f2c186de
    Signature Algorithm: sha256WithRSAEncryption
         7e:9e:ac:f4:63:99:0a:d3:6b:07:3c:e2:3e:e6:86:0a:b5:0a:
         b3:04:84:af:d1:fb:04:2d:1a:97:f5:f2:23:03:fe:e2:f3:35:
         17:6f:01:53:75:89:ac:a5:22:80:9d:63:e1:c2:38:aa:b9:64:
         f8:9c:ec:e9:cd:17:77:4e:b0:d5:06:99:3a:cc:c4:43:d8:5a:
         99:9f:12:fd:24:6f:e3:79:89:89:7c:1e:f1:8b:81:5e:c7:fc:
         15:25:f1:8f:12:55:b6:3c:03:23:24:3a:83:dc:eb:f1:52:34:
         d6:d0:87:1c:ce:68:0e:e5:51:9a:5b:f6:3c:5e:31:f2:9c:a0:
         51:6c:4d:8f:79:2b:c4:a0:14:f5:7d:1d:6c:c6:66:d5:b6:27:
         df:78:8e:0c:43:68:a9:86:55:81:2e:e1:4f:03:25:c8:0c:03:
         f2:8f:f0:2f:c6:f9:b2:a0:7d:d6:fb:23:8f:d9:cd:7b:a6:d2:
         cb:17:65:66:ea:66:62:bd:00:10:cb:93:60:09:fe:d9:da:d8:
         bb:3a:a4:ff:85:fa:4b:4e:28:e0:d1:f0:45:8a:aa:45:0a:b9:
         35:d0:4d:a0:75:ff:17:01:9d:86:5b:32:4f:50:88:ca:bb:83:
         8a:41:ab:34:40:1b:5d:33:43:5a:17:8a:16:2b:60:4a:2d:ab:
         5e:4a:4e:56
```

This would explain why I don't see traces from the Jaeger UI being shown there :-)

Reply:
Changed to use the headless service.

Reply:
Thanks, and sorry for the spam. GitHub was having some problems yesterday and it apparently added one comment per attempt.